plugins/outputs/file/README.md

```diff
@@ -41,4 +41,17 @@ See the [CONFIGURATION.md][CONFIGURATION.md] for more details.
   ## more about them here:
   ## https://github.com/influxdata/telegraf/blob/master/docs/DATA_FORMATS_OUTPUT.md
   data_format = "influx"
+
+  ## Compress output data with the specifed algorithm.
+  ## If empty, compression will be disabled and files will be plain text.
+  ## Supported algorithms are "zstd", "gzip" and "zlib".
+  # compression_algorithm = ""
+
+  ## Compression level for the algorithm above.
+  ## Please note that different algorithms support different levels:
+  ##   zstd  -- supports levels 1, 3, 7 and 11, default is 3.
+  ##   gzip -- supports levels -1, 0, 1 and 9, default is -1.
+  ##   zlib -- supports levels -1, 0, 1, and 9, default is -1.
```

Comment:
I don't think this is true. neither `gzip` nor `zlib` supports `-1` according to [internal/content_coding.go](https://github.com/influxdata/telegraf/blob/0291aafe514944245e6df42d7edfcb96101173a5/internal/content_coding.go#L187). So either the comment above or the implementation is wrong.

Reply:
The comment is wrong definitely. in fact the default level is -1.

https://pkg.go.dev/compress/flate#DefaultCompression

Reply:
And here comes my point. I don't care about what `compress/flate` says. We **ABSTRACTED** that level in `internal`. That was my whole point. My message said: if you do not set it, it will be the default (whatever that is!) but if you set it, it should be `1,3,7,11` or ` 0,1,9` respectively. Those values will then be **translated** in `internal` to the actual settings in `compress/flate` or whatever the underlying algorithm expects.